src/proxy/http/balance.rs

```diff
@@ -131,3 +133,69 @@ where
         Ok(Balance::p2c(loaded).into())
     }
 }
+
+pub mod weight {
+    use super::tower_balance::{HasWeight, Weight, Weighted};
+    use futures::{Future, Poll};
+    use svc;
+
+    #[derive(Clone, Debug)]
+    pub struct Layer(());
+
+    #[derive(Clone, Debug)]
+    pub struct MakeSvc<M> {
+        inner: M,
+    }
+
+    #[derive(Debug)]
+    pub struct MakeFuture<F> {
+        inner: F,
+        weight: Weight,
+    }
+
+    pub fn layer() -> Layer {
```

Comment:
TIOLI: It might be somewhat more concise to just use the [`svc::layer::mk` fn](https://github.com/linkerd/linkerd2-proxy/blob/master/src/svc.rs#L5), like how `pending` does:
https://github.com/linkerd/linkerd2-proxy/blob/61db2e77a247f7b0235b67581f60e8a92f8543cb/src/proxy/pending.rs#L23-L25

this avoids some of the boilerplate around having a layer type and impl that don't really do any _behavior_.